File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    let stack = 0;
    
    for(let i = 0; i < s.length; i++) {
        if(s[i] === '(') stack++;
        else {
            if(stack === 0) return false;
            stack--;
            // 빈 스택이면 return false, 아니라면 pop하고 넘어가기. 
        }
    }
    
    return !stack;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
from collections import deque

def solution(s):
    ls = []
    rs = deque(list(s))
    
    for i in range(len(s)):
        right_top = rs[0]
        
        if right_top == '(':
            ls.append(right_top)
            rs.popleft()
            
        elif right_top == ')':
            if len(ls) <= 0: 
                return False
            
            ls.pop()
            rs.popleft()
            
    if len(ls) == 0 and len(rs) == 0:
        return True
    
    return False
    
    
    
    